packages/flutter/lib/src/widgets/bottom_navigation_bar_item.dart

```diff
@@ -36,6 +38,13 @@ class BottomNavigationBarItem {
   /// [IconTheme] size and color.
   final Widget icon;
 
+  /// An alternative icon displayed when this bottom navigation item is
+  /// selected.
+  ///
+  /// If this icon is not provided, the bottom navigation bar will display
+  /// [icon] in either state.
```

Comment:
Can you add a paragraph about the motivation for providing a different icon? E.g. how you're supposed to use ativeIcon and icon to create accessible apps that don't just rely on color to differentiate between the two states.